languages/elixir/exercises/concept/streams/.docs/instructions.md

```diff
@@ -0,0 +1,38 @@
+While planning a games night with friends, the group decides they will revisit their childhood favorites. Your German friend proposes _Mensch ärgere Dich nicht_ (_Man, Don't Get Angry_), a classic in many European countries (similar to the English _Ludo_ or the North American _Parcheesi_).
+
+To prepare for the game, you decide to implement the dice rolling function in Elixir.
+
+In _Mensch ärgere Dich nicht_, players roll one 6-sided die to determine their next move. If they are lucky and roll 6, they can roll and move again.
+
+## 1. Do one 6-sided die roll
+
+Implement the `d6/0` function that will randomly choose one integer between 1 and 6 on each call.
+
+```elixir
+MenschAergereDichNicht.d6()
+# => 1
+MenschAergereDichNicht.d6()
+# => 5
+```
+
+## 2. Keep rolling as long as you roll sixes
+
+Implement the `roll/0` function. It should return a stream of dice rolls. The dice rolls should follow the rules of the game. If I roll a six, I can roll again. Otherwise, stop rolling. Use the `d6/0` function defined in the previous step.
+
+```elixir
+MenschAergereDichNicht.roll() |> Enum.to_list()
+# => [5]
+MenschAergereDichNicht.roll() |> Enum.to_list()
+# => [6, 3]
+```
+
+## 3. Allow using a rigged die
```

Comment:
This step feels a bit wrong because it has nothing to do with the concepts being taught (capturing functions and default arguments were already taught in previous exercises), but it is necessary, like Tim pointed out, to have a sane test suite.

Reply:
Agree and sympathize with your points.  Maybe a point could be made here about dependency injection patterns to make it less 'forced'?

Reply:
"You plan on expanding the game to use a 20-sided die, and decide to generalize the function"?

Reply:
As I mentioned in another comment, D&D makes heavy use of various sorts of dice: 6, 12, 20 amongst others.

Reply:
In the context of the story, swapping the die for a die with more sides doesn't make any sense :(. I was thinking maybe something "to play a joke on your friends...", but that sounds mean.

Unfortunately I have zero idea about D&D, otherwise I might have tried to some up with a D&D dice rolling story instead.

I'll keep it as it is for now.